backtrace-library/src/main/java/backtraceio/library/metrics/UniqueEvent.java

```diff
@@ -0,0 +1,52 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class UniqueEvent extends Event {
+
+    private final static transient String LOG_TAG = UniqueEvent.class.getSimpleName();
+
+    @SerializedName("unique")
+    private List<String> name;
+
+    UniqueEvent(String name) {
+        this(name, System.currentTimeMillis() / 1000, new HashMap<String, Object>());
+    }
+
+    UniqueEvent(String name, long timestamp, Map<String, Object> attributes) {
+        super(timestamp);
+        this.attributes = attributes;
+        this.name = new ArrayList<String>() {{
+            add(name);
+        }};
+    }
+
+    // The spec specifies name as a JSON array but we don't change the interface yet
+    @Override
+    public String getName() {
+        if (this.name != null && this.name.size() > 0) {
```

Comment:
Why do we store a list of strings instead of a single string? Let's make a step back: if this is a valid behavior if I have a list of strings and return as a name always the first one? 

In backtrace-unity, we don't have a list of strings that represents a unique event "name". The list of strings doesn't make sense in my opinion now. If it does - the "getName" method should return all strings, not a single one.

Reply:
I got confused because in the HTTP request we have to pass unique event "name" as a JSON array implying that unique event names should be an array. Should we assume unique event name will always be a JSON array of size 1? Maybe @vlussenburg can help out here too? 
https://docs.google.com/document/d/1_fCK4NHFiCjb4JmaO_R5drX8L4_OQhSneIuBdQ2z7Hs/edit#heading=h.7b32yhkhcc49

Reply:
From offline conversations we agree unique event should be treated as a single string, but unique events API spec still requires unique event names to be an array. I have added comments in the code explaining this discrepancy @konraddysput @vlussenburg